### PR TITLE
Edits to JOSS paper after review by Mark Harfouche

### LINKED ISSUE #131: JOSS Paper Review
xref: https://github.com/openjournals/joss-reviews/issues/1943

Looks good. 

* Scientficit software is also used to "Capture" or "Acquire" data
* "The usual" -> "One common approach" when talking about smaller datasets.
* The sentence that describes the challenge with including datasets in the github repo should come before the "Larger datasets [...]" sentence.
* HTTP -> HTTPS
* You state that people are recreating the same code, but maybe give a few examples?

Paragraph 2:
* [ ] It isn't clear what the registery holds: local file name, hash, remote file url should be clearly listed.
* [ ] I "minimal dependencies" is the selling point to packagers. The selling point to users is that it is easily installable with standard python distributinos PyPi and Conda, and a wide array of python versions (2.7, 3.5->3.8).

Paragraph 3:
* [ ] the functionality to unzip files is understated and should be made more explicit. It probably deserves a paragraph of its own.

Paragraph 4:
* [ ] You should elaborate on what exactly makes intake harder to setup. It seems like Pooch is "files only", while intake is "files + metadata" which is a larger hurdle to overcome.

Paragraph 5:
* [ ] I think you can show off a little bit explaining the motivation in terms of install size this would have on scikit-image. The numbers are listed in the PR

I guess I'll go through the checklist when the review officially starts.

### PR DESCRIPTION
**Pull request by leouieda**
Changes to the paper only to address comments by the reviewer. Mostly clarifying the advantages of Pooch.

Fixes #131 

openjournals/joss-reviews#1943

### CODE REVIEW

**Comment by leouieda** [comment]
Changes related to the comparison with Intake will be handled in another PR together with the review by @martindurant.

**Comment by leouieda** [comment]
Thanks, @matthewturk!

**Comment by leouieda** [comment]
Regarding the following comment:

> You state that people are recreating the same code, but maybe give a few examples?

I can think of Cartopy (which has code for downloading features) but would appreciate some help in finding other examples.

File: paper/paper.md [hunk 1]
```diff
@@ -52,29 +52,34 @@ bibliography: paper.bib
...
 The same problem is faced by scientists using version control to manage their
 research projects.
-As data files increase in size, it becomes unfeasible to store them in GitHub
-repositories.
-While downloading a data file over HTTP can be done easily with modern Python
+While downloading a data file over HTTPS can be done easily with modern Python
 libraries, it is not trivial to manage a set of files, keep them updated, and
 check for corruption.
+For example, scikit-learn [@scikit-learn], Cartopy [@cartopy], and PyVista
+[@pyvista] all include code dedicated to this particular task.
```
Comment (leouieda) [inline paper/paper.md]: Found these 3 examples. All include some form of download, caching, and hash validation.

**Comment by leouieda** [comment]
@hmaarrfk thank you for your comments! This PR addresses most of them. Please take a look to see if this is what you had in mind.

A few things that have not been added here:

> the functionality to unzip files is understated and should be made more explicit. It probably deserves a paragraph of its own.

I think this is a bit against the purpose of the JOSS paper (provide a high-level overview) and would be repeating the documentation. So I left this out. But I'm willing to write a bit more about it if you think that it should be included.

>  You should elaborate on what exactly makes intake harder to setup. It seems like Pooch is "files only", while intake is "files + metadata" which is a larger hurdle to overcome.
 
I'll add this in a follow-up PR addressing @martindurant's review.

> I think you can show off a little bit explaining the motivation in terms of install size this would have on scikit-image. The numbers are listed in the PR.

I'm a bit uncertain about this. I understand the motivation and cutting in half a package that is downloaded as often as scikit-image is a big plus for PyPI and conda-forge. But it feels a bit silly bragging about things in the order of 10s of megabytes. I'd be curious what the other authors think.

**Comment by leouieda** [comment]
Pinging @santisoler @remram44 @hugovk @matthewturk @jrleeman @danshapero @andersy005

**Comment by santisoler** [comment]
Thanks @hmaarrfk and @matthewturk for the review!

> I'm a bit uncertain about this. I understand the motivation and cutting in half a package that is downloaded as often as scikit-image is a big plus for PyPI and conda-forge. But it feels a bit silly bragging about things in the order of 10s of megabytes. I'd be curious what the other authors think.

I don't quite like the idea of advertising installation sizes quantitatively, but I agree that Pooch's lightness is one of its strong points. Maybe we could make more emphasis on the advantages of having a package like Pooch with minimal dependencies, both for end users and for package maintainers.

File: paper/paper.bib [hunk 2]
```diff
@@ -76,3 +76,25 @@ @misc{predictatops
   publisher = {Zenodo},
   year = {2019}
 }
+
+@manual{cartopy,
+  author = {{Met Office}},
+  title = {Cartopy: a cartographic python library with a Matplotlib interface},
```
Comment (hugovk) [inline paper/paper.bib]:
```suggestion
  title = {Cartopy: a cartographic Python library with a Matplotlib interface},
```

File: paper/paper.bib [hunk 3]
```diff
@@ -76,3 +76,25 @@ @misc{predictatops
...
+  year = {2010 - 2015},
+  address = {Exeter, Devon },
```
Comment (hugovk) [inline paper/paper.bib]:
```suggestion
  address = {Exeter, Devon},
```

Comment (hugovk) [inline paper/paper.bib]: Alternatively:
```suggestion
  address = {Exeter, UK},
```

File: paper/paper.bib [hunk 4]
```diff
@@ -76,3 +76,25 @@ @misc{predictatops
...
+  url = {http://scitools.org.uk/cartopy}
```
Comment (hugovk) [inline paper/paper.bib]:
```suggestion
  url = {https://scitools.org.uk/cartopy}
```

File: paper/paper.md [hunk 5]
```diff
@@ -52,29 +52,34 @@ bibliography: paper.bib
...
 Instead of scientists and library authors recreating the same code, it would be
 best to have a minimalistic and easy to set up tool for fetching and maintaining
 data files.
 
 Pooch is a Python library that fills this gap.
-It manages a data *registry* by downloading files from one or more remote
-servers and storing them in a local data cache.
+It manages a data *registry* (containing file names, SHA256 hashes, and
```
Comment (hugovk) [inline paper/paper.md]: Hyphenated looks more common, but if so, make sure to be consistent in the whole document.

```suggestion
It manages a data *registry* (containing file names, SHA-256 hashes, and
```

File: paper/paper.md [hunk 6]
```diff
@@ -52,29 +52,34 @@ bibliography: paper.bib
...
+download URLs) by downloading files from one or more remote servers and storing
+them in a local data cache.
 Pooch is written in pure Python and has minimal dependencies.
+It can be easily installed from the Python Package Index (PyPI) and conda-forge
+on a wide range of Python versions: 2.7 (up to Pooch 0.6.0) and from 3.5 to 3.8.
 The integrity of downloads is verified by comparing the file's SHA256 hash with
```
Comment (hugovk) [inline paper/paper.md]:
```suggestion
The integrity of downloads is verified by comparing the file's SHA-256 hash with
```

**Comment by leouieda** [comment]
@hugovk thanks for the suggestions :+1:  

> ...I agree that Pooch's lightness is one of its strong points. Maybe we could make more emphasis on the advantages of having a package like Pooch with minimal dependencies, both for end users and for package maintainers.

@santisoler I think the point was that using Pooch can allow other libraries to distribute smaller packages (because data are not included in the distribution). Maybe we could just mention this without quoting numbers.

File: paper/paper.md [hunk 7]
```diff
@@ -143,7 +148,9 @@ The Pooch API is stable and has been field-tested by other projects:
 MetPy [@metpy], Verde [@verde], RockHound [@rockhound], predictatops
 [@predictatops], and icepack [@icepack].
 Pooch is also being implemented as the download manager for scikit-image
-([GitHub pull request number 3945](https://github.com/scikit-image/scikit-image/pull/3945)).
+([GitHub pull request number 3945](https://github.com/scikit-image/scikit-image/pull/3945)),
+which will allow the project to use larger sample data while simultaneously
+reducing the download size of source and binary distributions.
```
Comment (leouieda) [inline paper/paper.md]: Added this sentence describing what scikit-image gets out of using Pooch.

File: paper/paper.md [hunk 8]
```diff
@@ -52,30 +52,35 @@ bibliography: paper.bib
...
+For example, scikit-learn [@scikit-learn], Cartopy [@cartopy], and PyVista
+[@pyvista] all include code dedicated to this particular task.
 Instead of scientists and library authors recreating the same code, it would be
 best to have a minimalistic and easy to set up tool for fetching and maintaining
 data files.
 
 Pooch is a Python library that fills this gap.
-It manages a data *registry* by downloading files from one or more remote
-servers and storing them in a local data cache.
+It manages a data *registry* (containing file names, SHA-256 hashes, and
```
Comment (hmaarrfk) [inline paper/paper.md]:
```suggestion
It manages a data *registry* (containing file names, cryptographic hashes -- e.g. sha256, and
```

**Comment by hmaarrfk** [comment]
Thanks these address many things! Looks great!

File: paper/paper.md (hunk 8 above)
Comment (leouieda) [reply paper/paper.md]: Thanks! I removed the "e.g." since we can only use SHA256 at the moment.

**Pull request merged**